chips/ibex/src/chip.rs

```diff
@@ -15,8 +15,12 @@ use crate::plic;
 use crate::timer;
 use crate::uart;
 
+#[cfg(target_device = "nexysvideo")]
 pub const CHIP_FREQ: u32 = 50_000_000;
 
+#[cfg(target_device = "verilator")]
+pub const CHIP_FREQ: u32 = 500_000;
```

Comment:
I agree. I was thinking to make a split something along the lines of what `boards/nordic` and others do. However it is slightly awkward as nexysvideo and verilator is the same Opentitan RTL. That also brings me to a separate question, whether in chips split into lowrisc and ibex is right, or should be done somewhat differently?

Reply:
> because things outside of the makefile flow end up with no `CHIP_FREQ` set

I think this is the primary issue: statements are not being generated. I think this problem can be circumvented or at least reduced by resorting to the `cfg!(...)` macro. I guess using something like this in the code should work:
```
let chip_freq =
    if cfg!(target_device = "verilator") {
        500_000
    } else {
        other_freq
    };
```

As every `if` is an expression, this forces you to provide a return value in _every_ case and shouldn't be able to cause compile errors (although you can of course deliberately panic or abort compilation in a branch). Even within runtime code, this would be inlined as to have to performance impact.